config/log_config.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/logging"
+)
+
+// LogPatternConfig is an instance of a level specification for a given logger.
+type LogPatternConfig struct {
+	Pattern string `json:"pattern"`
+	Level   string `json:"level"`
+}
+
+func validatePattern(pattern string) bool {
+	size := len(pattern)
+	if size == 0 || pattern[0] == '.' || pattern[size-1] == '.' {
+		return false
+	}
+	for i := 1; i < size; i++ {
+		if pattern[i] == '.' && pattern[i-1] == '.' {
+			return false
+		}
+	}
+	r := regexp.MustCompile(`^[a-z_-]+$|^\*$`)
+	for _, sep := range strings.Split(pattern, ".") {
+		if !r.MatchString(sep) {
```

Comment:
1) I think we can simplify the check a bit
2) we might as well compile the regexp once since it doesn't change

```suggestion
loggerPatternRegexp := regexp.MustCompile(`^[a-z_-]+$|^\*$`)

func validatePattern(pattern string) bool {
        for _, subp := range strings.Split(pattern, ".") {
                if !logConfigRegexp.MatchString(sep) {
                        return false
                }
        }
        return true
```